controllers/webhook.go

```diff
@@ -0,0 +1,189 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name: webhookServiceName,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "https",
+					Port:       443,
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.FromInt32(9443),
+				},
+			},
+			Selector: map[string]string{
+				"app.kubernetes.io/name": "odf-operator",
+			},
+			Type: corev1.ServiceTypeClusterIP,
+		},
+	}
+
+	csvWebhook = admrv1.MutatingWebhook{
+		Name: webhookName,
+		ClientConfig: admrv1.WebhookClientConfig{
+			Service: &admrv1.ServiceReference{
+				Name: webhookServiceName,
+				Path: ptr.To(WebhookPath),
+				Port: ptr.To(int32(443)),
+			},
+		},
+		Rules: []admrv1.RuleWithOperations{{
+			Rule: admrv1.Rule{
+				APIGroups:   []string{"operators.coreos.com"},
+				APIVersions: []string{"v1alpha1"},
+				Resources:   []string{"clusterserviceversions"},
+				Scope:       ptr.To(admrv1.NamespacedScope),
+			},
+			Operations: []admrv1.OperationType{admrv1.Create},
+		}},
+		ObjectSelector: &metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				{
+					Key:      "olm.copiedFrom",
+					Operator: metav1.LabelSelectorOpDoesNotExist,
+				},
+			},
+		},
+		SideEffects:             ptr.To(admrv1.SideEffectClassNone),
+		TimeoutSeconds:          ptr.To(int32(30)),
+		AdmissionReviewVersions: []string{"v1"},
+		// fail the admission if webhook can't be reached
+		FailurePolicy: ptr.To(admrv1.Fail),
+	}
+)
+
+func ensureWebhook(ctx context.Context, cli client.Client, logger logr.Logger, operatorNamespace string) error {
+
+	if err := reconcileCsvMutatingWebhookService(ctx, cli, logger, operatorNamespace); err != nil {
+		logger.Error(err, "unable to reconcile webhook service")
+		return err
+	}
+
+	if err := reconcileCsvMutatingWebhookConfiguration(ctx, cli, logger, operatorNamespace); err != nil {
+		logger.Error(err, "unable to register csv mutating webhook")
+		return err
+	}
+
+	return nil
+}
+
+func reconcileCsvMutatingWebhookService(ctx context.Context, cli client.Client, logger logr.Logger, operatorNamespace string) error {
+
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      webhookService.ObjectMeta.Name,
+			Namespace: operatorNamespace,
+		},
+	}
+	res, err := controllerutil.CreateOrUpdate(ctx, cli, svc, func() error {
+		if svc.Annotations == nil {
+			svc.Annotations = map[string]string{}
+		}
+		svc.Annotations["service.beta.openshift.io/serving-cert-secret-name"] = "odf-operator-webhook-server-cert"
+		webhookService.Spec.DeepCopyInto(&svc.Spec)
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	logger.Info("successfully created or updated webhook service", "operation", res, "name", svc.Name)
+	return nil
+}
+
+func reconcileCsvMutatingWebhookConfiguration(ctx context.Context, cli client.Client, logger logr.Logger, operatorNamespace string) error {
+
+	whConfig := &admrv1.MutatingWebhookConfiguration{}
+	whConfig.Name = csvWebhook.Name
```

Comment:
You have a constant, use it:
```suggestion
	whConfig.Name = webhookName
```

Reply:
constant is removed.